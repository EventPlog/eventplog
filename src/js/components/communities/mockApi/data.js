export default {
  communities: [
    {
      id: 1,
      name: 'Facebook Developer Circle Lagos',
      description: 'A community of developers united in knowledge sharing.',
      featured_image: '/login-bg.jpg',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Technology',
      joined: true,
    },
    {
      id: 2,
      name: 'Google Developer Group',
      description: 'Developer community bringing up community interest, powereed by google.',
      featured_image: '/tech_is_in_you.png',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Developers',
      joined: true,
    },
    {
      id: 3,
      name: 'Facebook Developer Circle Lagos',
      description: 'A community of developers united in knowledge sharing',
      featured_image: '/login-bg.jpg',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Technology',
      joined: true,
    },
    {
      id: 4,
      name: 'Google Developer Group',
      description: 'Developer community bringing up community interest, powereed by google.',
      featured_image: null,
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Developers',
      joined: false,
    },
    {
      id: 5,
      name: 'Facebook Developer Circle Lagos',
      description: 'A community of developers united in knowledge sharing',
      featured_image: '/tech_is_in_you.png',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Technology',
      joined: false,
    },
    {
      id: 6,
      name: 'Google Developer Group',
      description: 'Developer community bringing up community interest, powereed by google.',
      featured_image: null,
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Developers',
      joined: false,
    }
  ],

  events: [
    {
      id: 1,
      title: 'Tech is in you',
      featured_image: '/tech_is_in_you.png',
      interested_persons: 211,
      interested: false,
      community: {
        id: 1,
        name: 'Google Developer Group',
        description: 'Developer community bringing up community interest, powereed by google.',
        featured_image: '/login-bg.jpg',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Developers',
        joined: false,
      }
    },
    {
      id: 2,
      title: 'Writers corner workshop',
      featured_image: '/login-bg.jpg',
      interested_persons: 128,
      interested: false,
      community: {
        id: 2,
        name: 'Facebook Developer Circle Lagos',
        description: 'A community of developers united in knowledge sharing',
        featured_image: '/tech_is_in_you.png',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Technology',
        joined: false,
      }
    },
    {
      id: 3,
      title: 'Tech is in you',
      featured_image: '/tech_is_in_you.png',
      interested_persons: 221,
      interested: true,
      community: {
        id: 1,
        name: 'Google Developer Group',
        description: 'Developer community bringing up community interest, powereed by google.',
        featured_image: '/login-bg.jpg',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Developers',
        joined: false,
      }
    },
    {
      id: 4,
      title: 'Writers corner workshop',
      featured_image: '/login-bg.jpg',
      interested_persons: 18,
      interested: true,
      community: {
        id: 2,
        name: 'Facebook Developer Circle Lagos',
        description: 'A community of developers united in knowledge sharing',
        featured_image: '/tech_is_in_you.png',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Technology',
        joined: false,
      }
    }
  ]
}