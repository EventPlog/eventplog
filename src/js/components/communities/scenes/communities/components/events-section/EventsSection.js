import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'
import { genEventLink, genCommunityLink } from 'js/utils'

export const generateTitle = (event = {}, community = {}) => (
  <Link to={genEventLink(event, community)}>
    {event.title}
  </Link>
)

export const generateDescription = (community) => (
  <span>
    By <Link to={genCommunityLink(community)}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} people interested`
)


const EventsSection = ({ events }) => {
  const {loading, error, data } = events
  return (
    <Sidebar title="Events you may like">
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event, community);
          const description = community ? generateDescription(community) : '';
          const meta = generateMeta(event)
          return <Sidebar.Card
                    key={event.id}
                    {...{title, description, featured_image, meta}} />
        }
      )}
    </Sidebar>
  )
}

export default EventsSection