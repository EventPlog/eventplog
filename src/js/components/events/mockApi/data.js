export default {
  communities: [
    {
      id: 1,
      name: 'Facebook Developer Circle Lagos',
      description: 'A community of developers united in knowledge sharing.',
      featured_image: '/login-bg.jpg',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Technology',
      joined: true,
    },
    {
      id: 2,
      name: 'Google Developer Group',
      description: 'Developer community bringing up community interest, powereed by google.',
      featured_image: '/tech_is_in_you.png',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Developers',
      joined: true,
    },
    {
      id: 3,
      name: 'Facebook Developer Circle Lagos',
      description: 'A community of developers united in knowledge sharing',
      featured_image: '/login-bg.jpg',
      no_of_members: 200,
      no_of_upcoming_events: 2,
      focus: 'Technology',
      joined: true,
    },
  ],

  events: [
    {
      id: 1,
      title: 'Tech is in you',
      featured_image: '/tech_is_in_you.png',
      interested: true,
      interested_persons: 211,
      start_date: '28 June, 2018',
      start_time: '10:00am',
      end_date: '28 June, 2018',
      end_time: '01:00pm',
      venue: 'StartZone, Gbagada',
      community: {
        id: 3,
        name: 'Google Developer Group',
        description: 'Developer community bringing up community interest, powereed by google.',
        featured_image: '/login-bg.jpg',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Developers',
        joined: false,
      }
    },
    {
      id: 2,
      title: 'Writers corner workshop',
      featured_image: '/login-bg.jpg',
      interested: true,
      interested_persons: 211,
      start_date: '28 June, 2018',
      start_time: '10:00am',
      end_date: '28 June, 2018',
      end_time: '01:00pm',
      venue: 'StartZone, Gbagada',
      community: {
        id: 1,
        name: 'GDG Lagos',
        description: 'A community of developers united in knowledge sharing',
        featured_image: '/tech_is_in_you.png',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Technology',
        joined: false,
      }
    },
    {
      id: 3,
      title: 'Figma Design Series',
      featured_image: '/tech_is_in_you.png',
      interested: false,
      interested_persons: 108,
      start_date: '28 June, 2018',
      start_time: '10:00am',
      end_date: '28 June, 2018',
      end_time: '01:00pm',
      venue: 'StartZone, Gbagada',
      community: {
        id: 2,
        name: 'Facebook Developer Circle Lagos',
        description: 'A community of developers united in knowledge sharing',
        featured_image: '/tech_is_in_you.png',
        no_of_members: 200,
        no_of_upcoming_events: 2,
        focus: 'Technology',
        joined: false,
      }
    }
  ]
}