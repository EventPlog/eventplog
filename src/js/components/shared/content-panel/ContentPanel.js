import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { lighten } from 'polished'

// internal
import { media, maxMedia } from 'js/styles/mixins'
import ContentPanelCard from './ContentPanelCard'


const StyledContentPanel = styled.div`
  background: white;
  padding: 1rem;
  box-shadow: 1px 2px 4px ${props => lighten(0.45, props.theme.activeLink)};
    
  ${
    media.phone`
      max-width: 100%;
      padding: 0;
      box-shadow: none;
    `
  }
  .content-header {
    border-bottom: 1px solid ${props => lighten(0.2, props.theme.blue)};
    font-size: 2rem;
    font-weight: 900;
    letter-spacing: 0.01rem;
    margin: 0 1rem;
    padding-bottom: 5px;
    
    ${
      maxMedia.tablet`
        margin: 0 0 1rem;
      `
    }
  }
  
  > div.content-body {
    margin: 2rem;
    
    ${
      media.tablet`
        margin: 0 0 1rem;
      `
    }
        
    ${
      media.phone`
        margin: 0 0 1rem;
      `
    }
  }
  
`

const ContentPanel = function({
  className = '',
  title,
  style = {},
  bodyStyle = {},
  children
}) {
  return (
    <StyledContentPanel style={style} className={`${className} content-panel`}>
      {title && <h5 className="content-header">{ title }</h5>}
      <div className="content-body" style={bodyStyle}>
        { children }
      </div>
    </StyledContentPanel>
  )
}

ContentPanel.Card = ContentPanelCard

export default ContentPanel