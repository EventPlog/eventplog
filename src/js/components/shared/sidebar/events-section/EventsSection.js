import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'

export const generateTitle = (event = {}) => (
  <Link to={`/communities/${event.community_id}/events/${event.id}`}>
    {event.title}
  </Link>
)

export const generateDescription = (community) => (
  <span>
    By <Link to={`/communities/${community.id}`}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} people interested`
)


const EventsSection = ({
  title,
  events = {data: [], meta: {}},
  attendEvent
}) => {
  const {loading, error, data = {} } = events
  return (
    <Sidebar title={title || "Events you may like"}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event);
          const description = generateDescription(community);
          const meta = generateMeta(event)
          const btn = event.is_attending
                      ? {}
                      : {onClick: () => attendEvent(event), text: 'Interested'}
          return (
            <Link to={`/communities/${event.community_id}/events/${event.id}`}>
              <Sidebar.Card key={event.id}
                             {...{title, description,
                             featured_image, meta, btn}} />
            </Link>
          )
        }
      )}
    </Sidebar>
  )
}

export default EventsSection