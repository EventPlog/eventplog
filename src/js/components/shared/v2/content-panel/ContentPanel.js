import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { lighten } from 'polished'

// internal
import { media, maxMedia } from 'js/styles/mixins'
import ContentPanelCard from './ContentPanelCard'


const StyledContentPanel = styled.div`
  ${
    media.phone`
      max-width: 100%;
      padding: 0;
      box-shadow: none;
    `
  }
  .content-header {
    margin: 0;
  }

    ${
      maxMedia.tablet`
        margin: 0 0 1rem;
      `
    }
  }
  
  > div.content-body {
    // margin: 2rem 1rem;
    
    ${
      media.tablet`
        margin: 0 0 1rem;
      `
    }
        
    ${
      media.phone`
        margin: 0 0 1rem;
      `
    }
  }
  
`

const ContentPanel = function({
  className = '',
  title,
  style = {},
  bodyStyle = {},
  children
}) {
  return (
    <StyledContentPanel style={style} className={`${className} content-panel`}>
      {title && <h4 className="content-header">{ title }</h4>}
      <div className="content-body" style={bodyStyle}>
        { children }
      </div>
    </StyledContentPanel>
  )
}

ContentPanel.Card = ContentPanelCard

export default ContentPanel
