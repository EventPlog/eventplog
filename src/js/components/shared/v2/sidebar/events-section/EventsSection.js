import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'
import { pluralize, genCommunityLink, genEventLink } from 'js/utils'

export const generateTitle = (event = {}, community = {}) => (
  <Link to={genEventLink(event, community)}>
    {event.title}
  </Link>
)

export const generateDescription = (community = {}) => (
  <span>
    By <Link to={genCommunityLink(community)}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} ${pluralize('person', event.interested_persons)} interested`
)


const EventsSection = ({
  title,
  events = {data: [], meta: {}},
  attendEvent
}) => {
  const {loading, error, data = [] } = events
  return (
    <Sidebar title={title || "Events you may like"}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event, community);
          const description = community ? generateDescription(community) : '';
          const meta = generateMeta(event)
          const titleLink = genEventLink(event, community)
          const btn = event.is_attending
                      ? {}
                      : {onClick: () => attendEvent(event), text: 'Interested'}
          return (
            <Sidebar.Card key={event.id}
                           {...{title, description, titleLink,
                           featured_image, meta, btn}} />
          )
        }
      )}
    </Sidebar>
  )
}

export default EventsSection