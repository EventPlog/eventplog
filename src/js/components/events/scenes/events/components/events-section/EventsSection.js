import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

// internal
import ContentPanel from 'js/components/shared/content-panel'
import Loading from 'js/components/shared/loading'
import Error from 'js/components/shared/loading/Error'
import Button from 'js/components/shared/button'
import Pagination from 'js/components/shared/pagination'
import { pluralize } from 'js/utils'

export const generateTitle = (event = {}) => {
  const community = event.community || {}
  return (
    <Link to={`/communities/${event.community_id}/events/${event.id}`}>
      {event.title}
    </Link>
  )
}

export const generateDescription = (community = {}) => (
  <span>
    By <Link to={`/communities/${community.id}`}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event = {}) => ([
  <ul key={`date${event.id}`}>
    <li>
      {event.date}
    </li>
    <li>
      {event.time}
    </li>
    <li>
      {event.venue}
    </li>
  </ul>,
  <ul key={`interest${event.id}`}>
    <li>
      {event.interested_persons} {pluralize('person', event.interested_persons)} interested
    </li>
  </ul>
])

export const generateCTA = (handleClick) => (
  <Button onClick={handleClick}>
    Interested
  </Button>
)

const EventsSection = ({
  title,
  events = {},
  attendEvent,
  getEvents,
}) => {
  const {loading, error, data = [], meta = {}} = events;
  const shouldDisplayData = (!loading && !error && data);
  return (
    <ContentPanel title={title}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {shouldDisplayData && data.map(({featured_image, ...event}) => {
          const title = generateTitle(event)
          const description = generateDescription(event.community)
          const meta = generateMeta(event)
          const btn = {onClick: () => attendEvent(event), text: 'interested'}
          const titleLink = `/communities/${event.community_id}/events/${event.id}`

          return (
            <ContentPanel.Card
              key={event.id}
              {...{title, description, featured_image, meta, btn, titleLink}}
              showButton={!event.is_attending} />
          )
        }
      )}
      {shouldDisplayData && data.length < 1 && <p>No events to display today ...</p>}
      {
        meta && meta.total_pages && (data.length > 0 || meta.current_page > 1)
          ? <Pagination totalPages={meta.total_pages}
                        activePage={meta.current_page}
                        onPageChange={getEvents} />
          : ''
      }
    </ContentPanel>
  )
}

export default EventsSection