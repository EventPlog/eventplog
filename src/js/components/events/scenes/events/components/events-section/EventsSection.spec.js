import React from 'react';
import EventsSection, {
  generateTitle,
  generateDescription,
  generateMeta
} from './EventsSection';
import { shallow } from 'enzyme';

import ContentPanel from 'js/components/shared/content-panel'
import data from 'js/mock-api/data'

describe('Communities::Communities::EventsSection', () => {
  const title = 'Events Section'
  const { events } = data

  it('should render correctly', () => {
    const wrapper = shallow( <EventsSection title={title} events={events} /> );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(ContentPanel).length).toEqual(1);
    expect(wrapper.find(ContentPanel).props().title).toEqual(title)
    expect(wrapper.find(ContentPanel.Card).length).toEqual(events.data.length);
  })

  it('should pass the right props to children', () => {
    const wrapper = shallow( <EventsSection title={title} events={events} /> );

    const controlPanelCardInstances = wrapper.find(ContentPanel.Card)
    const event = events.data[0]

    expect(controlPanelCardInstances.at(0).props().title).toEqual(generateTitle(event, event.community))
    expect(controlPanelCardInstances.at(0).props().description).toEqual(generateDescription(event.community))
    expect(controlPanelCardInstances.at(0).props().featured_image).toEqual(event.featured_image)
    expect(controlPanelCardInstances.at(0).props().meta).toEqual(generateMeta(event))
  })

  describe('when data is not complete', () => {
    it('should render without crashing', () => {
      const shabbyEvents = {
        data: [{id: 1}],
        meta: {}
      }

      const wrapper = shallow(<EventsSection title={title} events={shabbyEvents} />)

      expect(wrapper.find(ContentPanel).length).toEqual(1)
    })
  })
});