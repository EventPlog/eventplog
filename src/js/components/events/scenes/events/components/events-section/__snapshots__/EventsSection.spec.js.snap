// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Communities::Communities::EventsSection should render correctly 1`] = `
<ContentPanel
  title="Events Section"
>
  <Link
    replace={false}
    to="/communities/undefined/events/1"
  >
    <MainContentCard
      btn={
        Object {
          "onClick": [Function],
          "text": "interested",
        }
      }
      description={
        <span>
          By 
          <Link
            replace={false}
            to="/communities/1"
          >
            Facebook Developer Circle Lagos
          </Link>
        </span>
      }
      featured_image="/tech_is_in_you.png"
      key="1"
      meta={
        Array [
          <ul>
            <li />
            <li />
            <li>
                StartZone, Gbagada
            </li>
        </ul>,
          <ul>
            <li>
                211
                 people interested
            </li>
        </ul>,
        ]
      }
      showButton={true}
      title={
        <Link
          replace={false}
          to="/communities/undefined/events/1"
        >
          Tech is in you
        </Link>
      }
    />
  </Link>
  <Link
    replace={false}
    to="/communities/undefined/events/2"
  >
    <MainContentCard
      btn={
        Object {
          "onClick": [Function],
          "text": "interested",
        }
      }
      description={
        <span>
          By 
          <Link
            replace={false}
            to="/communities/2"
          >
            Google Developer Group
          </Link>
        </span>
      }
      featured_image="/login-bg.jpg"
      key="2"
      meta={
        Array [
          <ul>
            <li />
            <li />
            <li>
                StartZone, Gbagada
            </li>
        </ul>,
          <ul>
            <li>
                211
                 people interested
            </li>
        </ul>,
        ]
      }
      showButton={true}
      title={
        <Link
          replace={false}
          to="/communities/undefined/events/2"
        >
          Writers corner workshop with writing and use of English
        </Link>
      }
    />
  </Link>
  <Link
    replace={false}
    to="/communities/undefined/events/3"
  >
    <MainContentCard
      btn={
        Object {
          "onClick": [Function],
          "text": "interested",
        }
      }
      description={
        <span>
          By 
          <Link
            replace={false}
            to="/communities/3"
          >
            ForLoop Lagos
          </Link>
        </span>
      }
      featured_image="/login-bg.jpg"
      key="3"
      meta={
        Array [
          <ul>
            <li />
            <li />
            <li>
                StartZone, Gbagada
            </li>
        </ul>,
          <ul>
            <li>
                108
                 people interested
            </li>
        </ul>,
        ]
      }
      showButton={true}
      title={
        <Link
          replace={false}
          to="/communities/undefined/events/3"
        >
          Figma Design Series
        </Link>
      }
    />
  </Link>
</ContentPanel>
`;
